src/components/PresenceMap/component.js

```diff
@@ -0,0 +1,194 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import getMapContainerStyles from './getMapContainerStyles';
+import StyledContainer from './styled/container';
+import StyledMapContainer from './styled/mapContainer';
+
+const MAX_ZOOM = 14;
+
+export default function MapComponent(props) {
+    const { showTraffic, showTransit, center, zoom, objects, style } = props;
+    const container = useRef();
+    const mapContainer = useRef();
+    const [map, setMap] = useState(null);
+
+    const prevIsScriptLoaded = useRef(props.isScriptLoaded);
+    useEffect(() => {
+        if (!prevIsScriptLoaded.current && props.isScriptLoaded && props.isScriptLoadSucceed) {
+            // eslint-disable-next-line
+            initMap();
+        }
+        // eslint-disable-next-line
+    }, [props.isScriptLoaded, props.isScriptLoadSucceed]);
```

Comment:
this eslint exception is not needed if you add initMap as dependency, but like above it need to be declared before using it